test-app/src/main/java/org/readium/r2/testapp/utils/extensions/Uri.kt

```diff
@@ -6,22 +6,46 @@
 
 package org.readium.r2.testapp.utils.extensions
 
+import android.content.ContentResolver
 import android.content.Context
 import android.net.Uri
+import android.provider.MediaStore
 import java.io.File
 import java.util.*
 import org.readium.r2.shared.error.Try
 import org.readium.r2.testapp.utils.ContentResolverUtil
+import org.readium.r2.testapp.utils.tryOrNull
 
 suspend fun Uri.copyToTempFile(context: Context, dir: File): Try<File, Exception> =
     try {
         val filename = UUID.randomUUID().toString()
-        val extension = path
-            ?.let { File(it).extension }
-            ?: "tmp"
-        val file = File(dir, "$filename.$extension")
+        val file = File(dir, "$filename.${extension(context)}")
```

Comment:
The `extension` was missing from the URL in the case of a `content` scheme. This made the format sniffing more expensive.